File: mobile/src/models/ExecutiveFundHistory.ts
import { NavPoint } from "./NavPoint";
import { VolatilityPoint } from "./VolatilityPoint";
import { PredictionPoint } from "./PredictionPoint";

export interface ExecutiveFundHistory {
    nav: NavPoint[];

    volatility: VolatilityPoint[];

    prediction_history: PredictionPoint[];
}

File: app/view_models/executive_fund_history.rb
# frozen_string_literal: true

class ExecutiveFundHistory
  attr_reader :nav,
              :volatility,
              :prediction_history

  def initialize(
    nav:,
    volatility:,
    prediction_history:
  )
    @nav = nav
    @volatility = volatility
    @prediction_history = prediction_history

    freeze
  end
end